scripts/pre_push_hook.py

```diff
@@ -295,21 +298,41 @@ def _install_hook():
         print 'Copied file to .git/hooks directory'
 
 
-def does_diff_include_js_files(files_to_lint):
-    """Returns true if diff includes JS files.
+def does_diff_include_js_and_ts_files(files_to_lint):
+    """Returns true if diff includes JS and TS files.
 
     Args:
         files_to_lint: list(str). List of files to be linted.
 
     Returns:
-        bool. Status of JS files in diff.
+        bool. Status of JS and TS files in diff.
     """
 
     js_files_to_check = [
         filename for filename in files_to_lint if
         filename.endswith('.js')]
 
-    return bool(js_files_to_check)
+    extra_js_files = []
+    for filename in js_files_to_check:
+        if filename.startswith(('core', 'extensions')) and (
+                filename not in build.JS_FILEPATHS_NOT_TO_BUILD):
+            extra_js_files.append(filename)
+
+    if len(extra_js_files):
+        err_msg = (
```

Comment:
I'm not sure whether it's a proper place to do this check. (As the name of the function doesn't indicate this functionality.)

Reply:
Shall I add a new function which tests this behavior and is called by this function?

Reply:
I'm not sure whether we should do this kinda check in pre_push_hook.py, @apb7 Do you have any suggestion on this?

Reply:
Maybe we can create `does_diff_includes_frontend_scripts` which will check for .ts files and frontend .js files if found returns true.
Whereas in pre_commit_linter.py we can put this test where we are testing if the js file should be ts file.

I'm not sure about what patterns/layers we have for our linters. I'm asking this as I use to run pre_commit_linet before every commit so I expect all lint checks should be done in the pre_commit_linters. Leaving this to @apb7 to finalize the place for this test.

Reply:
@apb7, Ping!

Reply:
I agree with @DubeySandeep here -- all checks should be placed in the pre-commit script. The general use of the pre-commit script should be, as the name suggests, to run before every commit. Thus, we need to include this check there.

>Maybe we can create does_diff_includes_frontend_scripts which will check for .ts files and expected frontend .js files, if found returns true.
Whereas in pre_commit_linter.py we can put this test where we are testing whether the js file should be ts file.

This sounds perfectly fine. Thanks!